agent/proxycfg/testing.go

```diff
@@ -1506,6 +1507,31 @@ func TestConfigSnapshotMeshGatewayUsingFederationStates(t testing.T) *ConfigSnap
 	return testConfigSnapshotMeshGateway(t, true, true)
 }
 
+func TestConfigSnapshotMeshGatewayNewerInformationInFederationStates(t testing.T) *ConfigSnapshot {
```

Comment:
Do we have a test that validate the other alternative?  If not it would be good to have a test that verify switching from a data source to the other work properly.

Reply:
Great point! Have added ✅